collector/pt_heartbeat.go

```diff
@@ -0,0 +1,95 @@
+// Scrape pt-heartbeat data.
+
+package collector
+
+import (
+	"database/sql"
+	"fmt"
+	"strconv"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	// Metric subsystem
+	ptHeartbeat = "pt_heartbeat"
+	// pt-heartbeat query
+	// the second column allows gets the server timestamp at the exact same time the query is run
+	ptHeartbeatQuery = "SELECT UNIX_TIMESTAMP(ts), UNIX_TIMESTAMP(CONCAT(DATE(NOW()), ' ', CURTIME(6))), server_id from %s"
+)
+
+// Metric descriptors.
+var (
+	PtHeartbeatStoredDesc = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, ptHeartbeat, "stored_timestamp_seconds"),
+		"Timestamp stored in the pt-heartbeat table.",
+		[]string{"server_id"}, nil,
+	)
+	PtHeartbeatNowDesc = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, ptHeartbeat, "now_timestamp_seconds"),
+		"Timestamp of the current server.",
+		[]string{"server_id"}, nil,
+	)
+	PtHeartbeatDelayDesc = prometheus.NewDesc(
```

Comment:
This metric isn't necessary (and against Prometheus design) since you can do this in PromQL.